backend/wmg/data/tissue_mapper.py

```diff
@@ -0,0 +1,251 @@
+        """
+        Returns the label from and ontology term id that is in writable form
+        Example: "UBERON:0002048" returns "lung"
+        Example: "UBERON_0002048" raises ValueError because the ID is not in writable form
+        """
+
+        if ontology_term_id in self._cached_labels:
+            return self._cached_labels[ontology_term_id]
+
+        entity = self._get_entity_from_id(self.make_id_readable(ontology_term_id))
+        if entity:
+            result = entity.label[0]
+        else:
+            result = ontology_term_id
+
+        self._cached_labels[ontology_term_id] = result
+        return result
+
+    @staticmethod
+    def make_id_readable(ontology_term_id: str) -> str:
+        """
+        Converts ontology term id string from "UBERON:0002048" to "UBERON_0002048"
+        """
+        if ontology_term_id.count(":") != 1:
+            raise ValueError(f"{ontology_term_id} is an invalid ontology term id, it must contain exactly one ':'")
+        return ontology_term_id.replace(":", "_")
+
+    @staticmethod
+    def make_id_writable(ontology_term_id: str) -> str:
+        """
+        Converts ontology term id string from "UBERON_0002048" to "UBERON:0002048"
+        """
+        if ontology_term_id.count("_") != 1:
+            raise ValueError(f"{ontology_term_id} is an invalid ontology term id, it must contain exactly one '_'")
+        return ontology_term_id.replace("_", ":")
+
+    def _list_ancestors(self, entity: owlready2.entity.ThingClass, ancestors: List[str] = []) -> List[str]:
+        """
+        Recursive function that given an entity of an ontology, it traverses the ontology and returns
+        a list of all ancestors associated with the entity.
+        """
+
+        if self._is_restriction(entity):
+            # Entity is a restriction, check for part_of relationship
+
+            prop = entity.property.name
+            if prop != "BFO_0000050":
+                # BFO_0000050 is "part of"
+                return ancestors
+            ancestors.append(entity.value.name.replace("obo.", ""))
+
+            # Check for ancestors of restriction
+            self._list_ancestors(entity.value, ancestors)
+            return ancestors
+
+        elif self._is_entity(entity) and not self._is_and_object(entity):
+            # Entity is a superclass, check for is_a relationships
+
+            if entity.name in self.DENY_LIST:
+                return ancestors
+            ancestors.append(entity.name)
+
+            # Check for ancestors of superclass
+            for super_entity in entity.is_a:
+                self._list_ancestors(super_entity, ancestors)
+            return ancestors
+
+    def _get_entity_from_id(self, ontology_term_id: str) -> owlready2.entity.ThingClass:
+        """
+        Given a readable ontology term id (e.g. "UBERON_0002048"), it returns the associated ontology entity
+        """
+        # TODO: use the pinned ontology at `single-cell-curation`
```

Comment:
Does this change need to happen prior to merging into the feature branch?

Reply:
nope, waiting on Dave and Mim for final implementation of ontology traversing